[Site Isolation] Make more storage/indexeddb/modern tests pass on bot

szewai opened the pull request:
#### 932707eee9bb5f7afe49a057f30796e558f8a48b
<pre>
[Site Isolation] Make more storage/indexeddb/modern tests pass on bot
<a href="https://rdar.apple.com/148863016">rdar://148863016</a>
<a href="https://bugs.webkit.org/show_bug.cgi?id=291295">https://bugs.webkit.org/show_bug.cgi?id=291295</a>

Reviewed by Per Arne Vollan.

When running test in cross-origin iframe with site isolation enabled, the result dumping can be async: subframe process
sends message to UI process, UI process notifies main frame process, and main frame process dumps the result. This means
when site isolation is enabled, test script in subframe might proceed further, and this could lead dumped result to be
different from when site isolation is disabled and result dumping is sync in one process. For these specific tests, we
don&apos;t expect more events to be handled or more logs after all expected script code is executed (notifyDone() is called),
so we can just remove the event listener and avoid logging after tests are marked as ended.

* LayoutTests/platform/mac-site-isolation/TestExpectations:
* LayoutTests/storage/indexeddb/modern/resources/deletedatabase-1.js:
(prepareDatabase.event.target.transaction.onabort):
(prepareDatabase.event.target.transaction.onerror):
(prepareDatabase):
(continueTest1.request.onerror):
(continueTest1.request.onupgradeneeded):
(continueTest1):
(continueTest2.versionTransaction.oncomplete):
(continueTest2.versionTransaction.onabort):
(continueTest2.versionTransaction.onerror):
(continueTest2.request.onupgradeneeded):
(continueTest2.request.onsuccess):
(continueTest2.request.onerror):
(continueTest2):
(done): Deleted.
(log): Deleted.
* LayoutTests/storage/indexeddb/modern/resources/deleteindex-4.js:
(prepareDatabase.tx.onabort):
(prepareDatabase.tx.onerror):
(prepareDatabase.tx.oncomplete):
(finishTest): Deleted.
* LayoutTests/storage/indexeddb/modern/resources/double-abort.js:
(prepareDatabase.versionTransaction.onabort):
(prepareDatabase.versionTransaction.oncomplete):
(prepareDatabase.versionTransaction.onerror):
(prepareDatabase):
(done): Deleted.
* LayoutTests/storage/indexeddb/modern/resources/idbdatabase-deleteobjectstore-failures.js:
(prepareDatabase.versionTransaction.onabort):
(prepareDatabase.versionTransaction.onerror):
(prepareDatabase):
(continueTest1.transaction.onabort):
(continueTest1.transaction.onerror):
(continueTest1):
(continueTest2.openRequest.onerror):
(continueTest2.openRequest.onblocked):
(continueTest2.openRequest.onsuccess):
(continueTest2.versionTransaction.onabort):
(continueTest2.versionTransaction.oncomplete):
(continueTest2.versionTransaction.onerror):
(continueTest2.openRequest.onupgradeneeded):
(continueTest2):
(done): Deleted.
* LayoutTests/storage/indexeddb/modern/resources/idbdatabase-transaction-failures.js:
(prepareDatabase.request.onerror):
(prepareDatabase.versionTransaction.onabort):
(prepareDatabase.versionTransaction.onerror):
(prepareDatabase):
(continueTest):
(done): Deleted.
* LayoutTests/storage/indexeddb/modern/resources/idbtransaction-objectstore-failures.js:
(prepareDatabase.tx.onabort):
(prepareDatabase.tx.oncomplete):
(prepareDatabase.tx.onerror):
(prepareDatabase):
(done): Deleted.
* LayoutTests/storage/indexeddb/modern/resources/index-5.js:
(checkIndexValues.cursorRequest.onsuccess):
(prepareDatabase.versionTransaction.onabort):
(prepareDatabase.versionTransaction.oncomplete):
(prepareDatabase.versionTransaction.onerror):
(prepareDatabase):
(done): Deleted.
(log): Deleted.
* LayoutTests/storage/indexeddb/modern/resources/transaction-scheduler-1.js:
(prepareDatabase.request.onerror):
(prepareDatabase.versionTransaction.onabort):
(prepareDatabase.versionTransaction.onerror):
(prepareDatabase):
(request.onerror):
(request.onblocked):
(request.onupgradeneeded):
(setupRequest):
(request.onsuccess):
(transaction.onerror):
(transaction.onabort):
(transaction.oncomplete):
(startTransactionLoop):
(done): Deleted.
* LayoutTests/storage/indexeddb/modern/resources/transaction-scheduler-2.js:
(prepareDatabase.request.onerror):
(prepareDatabase.versionTransaction.onabort):
(prepareDatabase.versionTransaction.onerror):
(prepareDatabase):
(request.onerror):
(request.onblocked):
(request.onupgradeneeded):
(setupRequest):
(request.onsuccess):
(transaction.onerror):
(transaction.onabort):
(transaction.oncomplete):
(startTransactionLoop):
(done): Deleted.
* LayoutTests/storage/indexeddb/resources/shared.js:
(endTestWithLog):

Canonical link: <a href="https://commits.webkit.org/293596@main">https://commits.webkit.org/293596@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/5dd658b436242028e3677c7f1e37ba9052a2cbe3

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/99376 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/19025 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/9276 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/104507 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/49977 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/19314 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/27459 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/75630 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/32734 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/102383 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/14689 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/89717 "Passed tests") | [❌ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/55989 "Found 1 new API test failure: /WPE/TestSSL:/webkit/WebKitWebView/ephemeral-tls-errors (failure)") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/14483 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/7699 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/49337 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/84409 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/7786 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/106864 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/26490 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/19322 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/84592 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/26852 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/85920 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/84105 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/21332 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/28775 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/6461 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/20233 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/26430 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/26250 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/29563 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/27817 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

pvollan commented on LayoutTests/storage/indexeddb/modern/resources/idbdatabase-transaction-failures.js:
```diff
@@ -31,8 +27,8 @@ function prepareDatabase(event)
     versionTransaction.onabort = function(event) {
-        debug("versionchange transaction aborted");
-        done();
+        // Set event handler to null to avoid unexpected error message being printed.
```
Should event handler be set to null here?

szewai replied:
ah, actually I've moved the setting to line 12 -- I need to get rid of the comment here.

pvollan reviewed (approved):
R=me.

szewai committed:
[Site Isolation] Make more storage/indexeddb/modern tests pass on bot
rdar://148863016
https://bugs.webkit.org/show_bug.cgi?id=291295

Reviewed by Per Arne Vollan.

When running test in cross-origin iframe with site isolation enabled, the result dumping can be async: subframe process
sends message to UI process, UI process notifies main frame process, and main frame process dumps the result. This means
when site isolation is enabled, test script in subframe might proceed further, and this could lead dumped result to be
different from when site isolation is disabled and result dumping is sync in one process. For these specific tests, we
don't expect more events to be handled or more logs after all expected script code is executed (notifyDone() is called),
so we can just remove the event listener and avoid logging after tests are marked as ended.

* LayoutTests/platform/mac-site-isolation/TestExpectations:
* LayoutTests/storage/indexeddb/modern/resources/deletedatabase-1.js:
(prepareDatabase.event.target.transaction.onabort):
(prepareDatabase.event.target.transaction.onerror):
(prepareDatabase):
(continueTest1.request.onerror):
(continueTest1.request.onupgradeneeded):
(continueTest1):
(continueTest2.versionTransaction.oncomplete):
(continueTest2.versionTransaction.onabort):
(continueTest2.versionTransaction.onerror):
(continueTest2.request.onupgradeneeded):
(continueTest2.request.onsuccess):
(continueTest2.request.onerror):
(continueTest2):
(done): Deleted.
(log): Deleted.
* LayoutTests/storage/indexeddb/modern/resources/deleteindex-4.js:
(prepareDatabase.tx.onabort):
(prepareDatabase.tx.onerror):
(prepareDatabase.tx.oncomplete):
(finishTest): Deleted.
* LayoutTests/storage/indexeddb/modern/resources/double-abort.js:
(prepareDatabase.versionTransaction.onabort):
(prepareDatabase.versionTransaction.oncomplete):
(prepareDatabase.versionTransaction.onerror):
(prepareDatabase):
(done): Deleted.
* LayoutTests/storage/indexeddb/modern/resources/idbdatabase-deleteobjectstore-failures.js:
(prepareDatabase.versionTransaction.onabort):
(prepareDatabase.versionTransaction.onerror):
(prepareDatabase):
(continueTest1.transaction.onabort):
(continueTest1.transaction.onerror):
(continueTest1):
(continueTest2.openRequest.onerror):
(continueTest2.openRequest.onblocked):
(continueTest2.openRequest.onsuccess):
(continueTest2.versionTransaction.onabort):
(continueTest2.versionTransaction.oncomplete):
(continueTest2.versionTransaction.onerror):
(continueTest2.openRequest.onupgradeneeded):
(continueTest2):
(done): Deleted.
* LayoutTests/storage/indexeddb/modern/resources/idbdatabase-transaction-failures.js:
(prepareDatabase.request.onerror):
(prepareDatabase.versionTransaction.onabort):
(prepareDatabase.versionTransaction.onerror):
(prepareDatabase):
(continueTest):
(done): Deleted.
* LayoutTests/storage/indexeddb/modern/resources/idbtransaction-objectstore-failures.js:
(prepareDatabase.tx.onabort):
(prepareDatabase.tx.oncomplete):
(prepareDatabase.tx.onerror):
(prepareDatabase):
(done): Deleted.
* LayoutTests/storage/indexeddb/modern/resources/index-5.js:
(checkIndexValues.cursorRequest.onsuccess):
(prepareDatabase.versionTransaction.onabort):
(prepareDatabase.versionTransaction.oncomplete):
(prepareDatabase.versionTransaction.onerror):
(prepareDatabase):
(done): Deleted.
(log): Deleted.
* LayoutTests/storage/indexeddb/modern/resources/transaction-scheduler-1.js:
(prepareDatabase.request.onerror):
(prepareDatabase.versionTransaction.onabort):
(prepareDatabase.versionTransaction.onerror):
(prepareDatabase):
(request.onerror):
(request.onblocked):
(request.onupgradeneeded):
(setupRequest):
(request.onsuccess):
(transaction.onerror):
(transaction.onabort):
(transaction.oncomplete):
(startTransactionLoop):
(done): Deleted.
* LayoutTests/storage/indexeddb/modern/resources/transaction-scheduler-2.js:
(prepareDatabase.request.onerror):
(prepareDatabase.versionTransaction.onabort):
(prepareDatabase.versionTransaction.onerror):
(prepareDatabase):
(request.onerror):
(request.onblocked):
(request.onupgradeneeded):
(setupRequest):
(request.onsuccess):
(transaction.onerror):
(transaction.onabort):
(transaction.oncomplete):
(startTransactionLoop):
(done): Deleted.
* LayoutTests/storage/indexeddb/resources/shared.js:
(endTestWithLog):

Canonical link: https://commits.webkit.org/293596@main
